tests/system/python/e2e/test_e2e_vary_asset_http_pi.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test system/python/e2e/test_e2e_vary_asset_http_pi.py
+
+"""
+
+import http.client
+import json
+import time
+from datetime import datetime, timezone
+import uuid
+import pytest
+
+
+__author__ = "Vaibhav Singhal"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+class TestE2EAssetHttpPI:
+
+    @pytest.fixture
+    def start_south_north(self, reset_and_start_foglamp, add_south, start_north_pi_server_c, remove_directories,
+                          south_branch, foglamp_url, pi_host, pi_port, pi_token):
+        """ This fixture clone a south repo and starts both south and north instance
+            reset_and_start_foglamp: Fixture that resets and starts foglamp, no explicit invocation, called at start
+            add_south: Fixture that adds a south service with given configuration
+            start_north_pi_server_c: Fixture that starts PI north task
+            remove_directories: Fixture that remove directories created during the tests"""
+
+        south_plugin = "http"
+        add_south("http_south", south_branch, foglamp_url, config={"assetNamePrefix": {"value": ""}},
+                  service_name="http_south")
+        start_north_pi_server_c(foglamp_url, pi_host, pi_port, pi_token)
+
+        yield self.start_south_north
+
+        # Cleanup code that runs after the caller test is over
+        remove_directories("/tmp/foglamp-south-{}".format(south_plugin))
+
+    def test_end_to_end(self, start_south_north, read_data_from_pi, foglamp_url, pi_host, pi_admin, pi_passwd, pi_db,
+                        wait_time, retries):
+        """ Test that data is inserted in FogLAMP and sent to PI
+            start_south_north: Fixture that starts FogLAMP with south and north instance
+            read_data_from_pi: Fixture to read data from PI
+            Assertions:
+                on endpoint GET /foglamp/asset
+                on endpoint GET /foglamp/asset/<asset_name>
+                data received from PI is same as data sent"""
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        time.sleep(wait_time)
+
+        # Send data to foglamp-south-http
+        conn_http_south = http.client.HTTPConnection("localhost:6683")
+
+        asset_name = "e2e_varying"
+        sensor_data = [{"a": 1}, {"a": 2, "b": 3}, {"b": 4}]
```

Comment:
I suggest,  we should send sensor_data again after first retrieval verification!

Reply:
can the next data be like:

` sensor_data = [{"b": 1.1}, {"a2": 2, "b2": 3}, {"a": 4.0}]`

Reply:
Our current approach deletes web element from PI after reading, Instead I have added two list, 
```
        # 2 list having mixed data simulating different sensors
        # (sensors coming up and down, sensors throwing int and float data)
        sensor_data = [{"a": 1}, {"a": 2, "b": 3}, {"b": 4}]
        sensor_data_2 = [{"b": 1.1}, {"a2": 2, "b2": 3}, {"a": 4.0}]
```
which will replicate the desired scenario.